packages/engine/src/scene/systems/SceneLoadingSystem.test.tsx

```diff
@@ -0,0 +1,1380 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { EntityUUID } from '@etherealengine/common/src/interfaces/EntityUUID'
+
+import { applyIncomingActions, dispatchAction, getMutableState, getState } from '@etherealengine/hyperflux'
+import { act, render, waitFor } from '@testing-library/react'
+import assert from 'assert'
+import React from 'react'
+import { EditorControlFunctions } from '../../../../editor/src/functions/EditorControlFunctions'
+import { overrideFileLoaderLoad } from '../../../tests/util/loadGLTFAssetNode'
+import { EventDispatcher } from '../../common/classes/EventDispatcher'
+import { Engine, destroyEngine } from '../../ecs/classes/Engine'
+import { EngineState } from '../../ecs/classes/EngineState'
+import { SceneSnapshotAction, SceneSnapshotSystem, SceneState } from '../../ecs/classes/Scene'
+import { defineQuery, getComponent, hasComponent } from '../../ecs/functions/ComponentFunctions'
+import { EntityTreeComponent } from '../../ecs/functions/EntityTree'
+import { SystemDefinitions } from '../../ecs/functions/SystemFunctions'
+import { createEngine } from '../../initializeEngine'
+import { PhysicsState } from '../../physics/state/PhysicsState'
+import { SceneDataType, SceneID, SceneJsonType } from '../../schemas/projects/scene.schema'
+import { UserID } from '../../schemas/user/user.schema'
+import { FogSettingsComponent } from '../components/FogSettingsComponent'
+import { ModelComponent } from '../components/ModelComponent'
+import { NameComponent } from '../components/NameComponent'
+import { SceneAssetPendingTagComponent } from '../components/SceneAssetPendingTagComponent'
+import { UUIDComponent } from '../components/UUIDComponent'
+import { SceneLoadingSystem } from './SceneLoadingSystem'
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '1'
+
+const modelLink = '/packages/projects/default-project/assets/collisioncube.glb'
+const testScene = {
```

Comment:
let's put this in a .json file and import it to improve readability of this file

Reply:
roger that, ill put it right next to the file